test/libsolidity/smtCheckerTests/abi/abi_encode_with_selector_vs_sig.sol

```diff
@@ -8,5 +8,4 @@ contract C {
 	}
 }
 // ----
-// Warning 6328: (326-356): CHC: Assertion violation might happen here.
-// Warning 7812: (326-356): BMC: Assertion violation might happen here.
+// Warning 6328: (326-356): CHC: Assertion violation happens here.
```

Comment:
`might happen` means it couldn't solve it at all, so BMC tries. `happens` means it solved it and the property is unsafe, so BMC doesn't run

Reply:
Yes, we are still encountering this non-determinism in Spacer very often. Small change in the encoding causes the problem to solve something it could not solve before and the other way around. In this case, the change is positive, we can prove something which timed out before. But I think also the other, less desirable change occurred in another example.